src/Battle.php

```diff
@@ -0,0 +1,211 @@
+<?php
+declare(strict_types=1);
+
+namespace LotGD\Core;
+
+use LotGD\Core\{
+    DiceBag,
+    Exceptions\ArgumentException,
+    Exceptions\BattleIsOverException,
+    Exceptions\BattleNotOverException,
+    Models\FighterInterface
+};
+
+/**
+ * Description of Battle
+ */
+class Battle
+{
+    const DAMAGEROUND_PLAYER = 0b01;
+    const DAMAGEROUND_MONSTER = 0b10;
+    const DAMAGEROUND_BOTH = 0b11;
+    
+    protected $player;
+    protected $monster;
+    protected $diceBag;
+    protected $isOver = false;
+    protected $winner;
+    protected $looser;
+    
+    public function __construct(FighterInterface $player, FighterInterface $monster)
+    {
+        $this->player = $player;
+        $this->monster = $monster;
+        $this->diceBag = new DiceBag();
+    }
+    
+    public function getActions()
+    {
+        
+    }
+    
+    public function selectAction()
+    {
+        
+    }
+    
+    /**
+     * Returns true if the battle is over.
+     * @return type
+     */
+    public function isOver()
+    {
+        return $this->isOver;
+    }
+    
+    /**
+     * Returns the winner of this fight
+     * @return FighterInterface
+     */
+    public function getWinner(): FighterInterface
+    {
+        if (is_null($this->winner)) {
+            throw new BattleNotOverException('There is no winner yet.');
+        }
+        return $this->winner;
+    }
+    
+    /**
+     * Returns the looser of this fight
+     * @return FighterInterface
+     */
+    public function getLooser(): FighterInterface
+    {
+        if (is_null($this->looser)) {
+            throw new BattleNotOverException('There is no looser yet.');
+        }
+        return $this->looser;
+    }
+    
+    /**
+     * Fights the number of rounds given by the parameter $n and returns the number
+     * of actual rounds fought.
+     * @param int $n
+     * @param bool $firstDamageRound Which damage rounds are calculated. Cannot be 0.
+     * @return int Number of fights fought.
+     */
+    public function fightNRounds(int $n = 1, int $firstDamageRound = self::DAMAGEROUND_BOTH): int
+    {
+        if ($firstDamageRound === 0) {
+            throw new ArgumentException('$firstDamageRound must not be 0.');
+        }
+        
+        if ($this->isOver === true) {
+            throw new BattleIsOverException('This battle has already ended. You cannot fight anymore rounds.');
+        }
+        
+        for ($count = 0; $count < $n; $count++) {
+            $this->fightOneRound($firstDamageRound);
+            $isSurprised = self::DAMAGEROUND_BOTH;
+            
+            // If one of the participants is dead, abort.
+            if ($this->player->isAlive() === false || $this->monster->isAlive() === false) {
+                $this->isOver = true;
+                
```

Comment:
It can happen that both die (lets say, a buff kills the enemy, while the enemy counters the player's attack). That's why I check if one is dead. The player loses if he dies, no matter if the enemy is alive or not.